src/vcpkg/tools.cpp

```diff
@@ -609,13 +625,13 @@ coscli version v0.11.0-beta
         vcpkg::Cache<std::string, PathAndVersion> path_version_cache;
 
         ToolCacheImpl(Filesystem& fs,
-                      const DownloadManager& downloader,
+                      std::shared_ptr<const DownloadManager>&& downloader,
```

Comment:
```suggestion
                      const std::shared_ptr<const DownloadManager>& downloader,
```

No reason to block non-temporaries